### ISSUE TITLE
`vcpkg remove` does not update the status database after package removal

### ISSUE THREAD

**Issue by rkitover** [microsoft/vcpkg#13097]
****

**Is your feature request related to a problem? Please describe.**
The `Status:` field of a package does not get updated from `install ok installed` immediately after its removal, a subsequent operation such as `list` is required for the database to be updated.

**Proposed solution**
It should get updated immediately on a `remove`.

**Additional context**
Here is the log from my terminal illustrating the issue:

```powershell
✔ ~\src\vcpkg\installed\vcpkg [master ≡]
rkitover@epycwin > grep sfml -A 10 .\status
Package: sfml
Version: 2.5.1
Port-Version: 9
Depends: freetype, libflac, libogg, libvorbis, openal-soft, stb
Architecture: x64-windows-static
Multi-Arch: same
Abi: 8dc17b8325a06029c61f264bbfe0f10464b57f6b
Description: Simple and fast multimedia library
Type: Port
Status: install ok installed

✔ ~\src\vcpkg\installed\vcpkg [master ≡]
rkitover@epycwin > vcpkg remove sfml:x64-windows-static
Your feedback is important to improve Vcpkg! Please take 3 minutes to complete our survey by running: vcpkg contact --survey
The following packages will be removed:
    sfml:x64-windows-static
Removing package sfml:x64-windows-static...
Removing package sfml:x64-windows-static... done
✔ ~\src\vcpkg\installed\vcpkg [master ≡]
rkitover@epycwin > grep sfml -A 10 .\status
Package: sfml
Version: 2.5.1
Port-Version: 9
Depends: freetype, libflac, libogg, libvorbis, openal-soft, stb
Architecture: x64-windows-static
Multi-Arch: same
Abi: 8dc17b8325a06029c61f264bbfe0f10464b57f6b
Description: Simple and fast multimedia library
Type: Port
Status: install ok installed

✔ ~\src\vcpkg\installed\vcpkg [master ≡]
rkitover@epycwin > vcpkg list *> $null
✔ ~\src\vcpkg\installed\vcpkg [master ≡]
rkitover@epycwin > grep sfml -A 10 .\status
Package: sfml
Version: 2.5.1
Port-Version: 9
Depends: freetype, libflac, libogg, libvorbis, openal-soft, stb
Architecture: x64-windows-static
Multi-Arch: same
Abi: 8dc17b8325a06029c61f264bbfe0f10464b57f6b
Description: Simple and fast multimedia library
Type: Port
Status: purge ok not-installed

```

**Comment by JackBoosY** [comment]
@BillyONeal I think it's a vcpkg bug, or it's by design?

**Comment by BillyONeal** [comment]
Seems like a bug to me.

**Comment by ras0219** [comment]
This is not a bug: the status database is more nuanced than a single file.

Specifically, during execution, vcpkg writes update files that contain "deltas" to the status db. At any given point in time, the database should be considered as the _sum_ of the status file and all update files.

On launch, when loading the status db, vcpkg first performs a compacting step that computes the entire consolidated status file and writes that back out (atomically) to the status database. This is why you observe that running `vcpkg list` appears to "fix" the status file -- it performs a "gc" from many files down to one.

The reason to not write the database out on every update is that it would make installing N packages an O(N^2) operation -- every install step would need to re-write the binary paragraph of every install step before it. The reason to consolidate is to avoid needing to access O(N) files on _every_ execution.

tldr; not a bug, by design.

**Comment by rkitover** [comment]
Right so, what is the problem with updating the database after every **COMMAND**, if a command installs multiple packages it can still do that.

**Comment by BillyONeal** [comment]
>The reason to not write the database out on every update is that it would make installing N packages an O(N^2) operation -- every install step would need to re-write the binary paragraph of every install step before it. The reason to consolidate is to avoid needing to access O(N) files on every execution.

The compressed version of the database could be written as part of `vcpkg remove` without turning all operations into O(n^2).

**Comment by JackBoosY** [comment]
Duplicated to #10812.
